File: easy/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numsHash; // hash table where key is number, value is vector index of that number
        
        for (int i = 0; i < nums.size(); ++i) {
            if (numsHash.find(target - nums[i]) == numsHash.end()) // if number that added to the current number is not found in hash table
                numsHash[nums[i]] = i; // add current number and its index to hash table
            else
                return {numsHash[target - nums[i]], i}; // found needed value, return both indexes
        }
        
        return {};
    }
};

File: easy/Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsHash = {} # Key is a number, value is the number's index
        for i in range(0, len(nums)):
            if target - nums[i] not in numsHash.keys(): # Working "backwards" in a sense, target - nums[i] is the value we need to sum to target
                numsHash[nums[i]] = i # If we can't find the needed value, add the current value to the hash table and try again
            else:
                return [i, numsHash[target - nums[i]]] # If we do find the needed value, return the indexes of the current value and the matching value
